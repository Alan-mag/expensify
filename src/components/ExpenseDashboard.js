import React from "react";
import ExpenseList from './ExpenseList';
import ExpensesSummary from './ExpensesSummary';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboard = () => (
    <div>
        <ExpensesSummary></ExpensesSummary>
        <ExpenseListFilters></ExpenseListFilters>
        <ExpenseList></ExpenseList>
    </div>
);

export default ExpenseDashboard;