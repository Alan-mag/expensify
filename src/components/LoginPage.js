import React from 'react';
import { connect } from "react-redux";
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  // methods
  // render
  <div>
    <button onClick={startLogin}>Login</button>
  </div>
);

// map dispatch to props
const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

// export
export default connect(undefined, mapDispatchToProps)(LoginPage);