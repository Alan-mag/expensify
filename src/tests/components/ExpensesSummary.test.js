import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary Correctly', () => {
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={23234215} />);
  expect(wrapper).toMatchSnapshot();
});