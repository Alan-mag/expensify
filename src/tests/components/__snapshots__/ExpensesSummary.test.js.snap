// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h3>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    23
     
    expenses
     totalling 
    $232,342.15
  </h3>
</div>
`;

exports[`should render ExpensesSummary Correctly 1`] = `
<div>
  <h3>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h3>
</div>
`;
